tests/roles/dataplane_adoption/tasks/neutron_verify.yaml

```diff
@@ -33,5 +32,13 @@
     success_msg: "neutron-sriov-nic-agent is ALIVE after adoption"
 
 - name: verify connectivity to the existing test VM instance using Floating IP
-  ansible.builtin.shell: |
-      ping -c4 {{ lookup('env', 'FIP') | default('192.168.122.20', True) }}
+  block:
+    - name: Ping VM using Floating IP
+      ansible.builtin.shell: |
+        ping -c4 {{ lookup('env', 'FIP') | default('192.168.122.20', True) }}
+  rescue:
+    - name: Get console log of the test VM
+      ansible.builtin.shell: |
```

Comment:
maybe use failed_when = false for the above task, and register outputs, then the console log should use the fail module yielding the console log as an error message? See an example https://github.com/openstack-k8s-operators/edpm-ansible/pull/667/files#diff-14d838f079a0c024754d82e20d9e153b058aef1491302e4468d39584481be706R55-R63